api/translator/src/main/java/com/project/translator/adapter/out/persistence/MessageRepository.java

```diff
@@ -11,8 +11,11 @@ interface MessageRepository extends JpaRepository<MessageEntity, Long> {
 
     List<MessageEntity> findByTagsId(Long id);
 
-    @Query("select m from MessageEntity m join LanguageEntity l on m.language.id = l.id where l.language = :language")
-    List<MessageEntity> findByLanguage(String language);
+    @EntityGraph(attributePaths = {"language", "originalMessage", "tags"})
+    List<MessageEntity> findByLanguage_languageContainingIgnoreCase(String language);
+
+    @EntityGraph(attributePaths = {"language", "originalMessage", "tags"})
+    List<MessageEntity> findByTags_tagContainingIgnoreCase(String tag);
```

Comment:
W przypadku języka IgnoreCase jest zrozumiały, ale czy w przypadku taga też? W mojej opinii test, Test i TEST to mogą być 3 różne tagi. Do przedyskutowania na grupie

Reply:
Ciekawe spostrzeżenie, po dyskusji z teamem - usuwamy ignore case.